vulcanai2/models/cnn.py

```diff
@@ -92,8 +94,42 @@ def __init__(self, name, in_dim, config, save_path=None,
             lr_scheduler, early_stopping, criter_spec)
 
     def _create_network(self, **kwargs):
-
+        self._in_dim = self.in_dim
```

Comment:
Why do we have two of the same variables?

Reply:
this is essential to know for everyone, maybe we might change the usage in future, but as of now,
self.in_dim spits out dimensions of whatever input passed into a network, while self._in_dim (private) provides the dimension size of the actual input that get fed into the network
For example: if `self.in_dim = [(1,28,28), 784, (64, 2, 2)]`, then `self._in_dim = [(66, 28, 28)]`